framework/include/problems/SubProblem.h

```diff
@@ -1080,6 +1093,9 @@ class SubProblem : public Problem
   /// AD flag indicating whether **any** AD objects have been added
   bool _have_ad_objects;
 
+  /// the list of vector tags that the application developer wants to be responsible for zeroing
+  std::set<TagID> _not_zeroed_tagged_vectors;
```

Comment:
does this need to be ordered? If not, I suggest using `unordered_set`

Reply:
No there is no need for this to be an ordered set since we just need it to store ids and to check if they exist within the set. I will update this.